qp/ensemble.py

```diff
@@ -0,0 +1,519 @@
+        if samples is None:
+            self.samples = [None] * N
+        else:
+            self.samples = samples
+        if quantiles is None:
+            self.quantiles = [None] * N
+        else:
+            self.quantiles = [(quantiles[0], quantiles[1][i]) for i in self.pdf_range]
+        if histogram is None:
+            self.histogram = [None] * N
+        else:
+            self.histogram = [(histogram[0], histogram[1][i]) for i in self.pdf_range]
+        if gridded is None:
+            self.gridded = [None] * N
+        else:
+            self.gridded = [(gridded[0], gridded[1][i]) for i in self.pdf_range]
+        self.mix_mod = None
+        self.evaluated = None
+
+        self.scheme = scheme
+
+        if vb and self.truth is None and self.quantiles is None and self.histogram is None and self.gridded is None and self.samples is None:
+            print 'Warning: initializing an Ensemble object without inputs'
+            return
+
+        self.make_pdfs()
+
+        # self.logfilename = 'logfile.txt'
+
+        self.stacked = {}
+
+        # self.where = where
+        # if os.path.isfile(self.where):
+        #     #something like this, uses connection?
+        #     self.engine = self.read(self.where)
+        # else:
+        #     self.engine = sqlalchemy.create_engine('sql:///'+self.where)
+        #
+        # self.metadata = MetaData(bind=self.engine)
+        #
+        # parametrizations_table =Table('parametrizations', metadata,)
+
+    #
+    #
+    # def read(self):
+    #
+    # def write(self):
+    #
+    # def make_db(self, where):
+    #     """
+    #     Makes a fresh set of ensemble tables
+    #     """
+    #     return
+    #
+    # def __add__(self, ensemble):
+    #
+    # def add_PDFs(self, PDFs):
+    #     """
+    #     Adds qp.PDF objects to the ensemble
+    #
+    #     Parameters
+    #     ----------
+    #     PDFs: list, qp.PDF object
+    #         list of PDF objects to add to the ensemble
+    #     """
+    #     pdf_range = range(len(PDFs))
+    #     def add_one(i):
+    #
+    #
+    #     self.pdfs = self.pool.map(add_one, pdf_range)
+    #
+    #     return(self)
+    #
+    # def add_parameterization(self, type, parameters):
+    #
+
+    def make_pdfs(self):
+        """
+        Makes a list of qp.PDF objects based on input
+        """
+        def make_pdfs_helper(i):
+            # with open(self.logfilename, 'wb') as logfile:
+            #     logfile.write('making pdf '+str(i)+'\n')
+            return qp.PDF(truth=self.truth[i], quantiles=self.quantiles[i],
+                            histogram=self.histogram[i],
+                            gridded=self.gridded[i], samples=self.samples[i],
+                            scheme=self.scheme, vb=False)
+
+        self.pdfs = self.pool.map(make_pdfs_helper, self.pdf_range)
+        self.pdf_range = range(len(self.pdfs))
+
+        return
+
+    def sample(self, samps=100, infty=100., using=None, vb=True):
+        """
```

Comment:
Indeed!  All instances of `infty` in both `ensemble.py` and `pdf.py` now refer to the single magic number set in `utils.py`.